python/postprocessing/modules/jme/jetSmearer.py

```diff
@@ -168,6 +176,12 @@ def getSmearValsM(self, jetIn, genJetIn ):
 
         jet_m_sf_and_uncertainty = dict( zip( [enum_nominal, enum_shift_up, enum_shift_down], self.jmr_vals ) )
 
+        # Get mass resolution
```

Comment:
I think that makes sense but the change shouldn't occur here. Instead `jetSmearer.getSmearValsM()` should only be called in the case of "PUPPI" being in jetType in jetmetUncertainties.py. @camclean can you confirm?

Reply:
@lcorcodilos @camclean Let me try to write down what I think is here and you can correct me.  The method getSmearsValsM, in principle, could apply to CHS or PUPPI jets.  The current values in jmr_vals and jet_m_resolution are only applicable to PUPPI+SD jets with mass near the W mass.  If there is nothing in the current code preventing a user from calling getSmearValsM on non-PUPPI jets, this could cause confusion.  I think this confusion could be avoided by either checking jetType in getSmearValsM or renaming this method getSmearValsPuppiM (or something similar) and only calling it from jetmetUncertainties for PUPPI jets.  In this case, I think it would be better to check jetType in jetSmearValsM to avoid renaming methods/creating new methods for different jet types.  Am I misunderstanding? Thanks.

Reply:
The conclusion at the last JME meeting was that we should add a mass smearing option to jetmetUncertainties module. Then, we can set the default to off. We can add pre-fabricated modules with it set to off. I have been waiting for this PR to be concluded to implement these items.